fix graphite.db permissions & location & remove default apache config

JoshiiSinfield opened the pull request:
Apologies - I haven't had chance to do any more to this since the last commit, until now!

if installing from packages you get a default apache config -
graphite_web.conf. However, that has data in it which stops httpd from
starting therefore ensuring it's removed here as this module creates a
graphite.conf.

correcting the value of graphite.db's directory, to ensure that the
permissions are set on the correct file - in the case where the packages
set one up in /var/lib/graphite-web/graphite.db, and you specify this
value for the django backend name

dwerder commented on manifests/init.pp:
```diff
@@ -759,7 +759,7 @@
   $graphiteweb_conf_dir_REAL        = pick($gr_graphiteweb_conf_dir,   "${base_dir_REAL}/conf")
   $graphiteweb_webapp_dir_REAL      = pick($gr_graphiteweb_webapp_dir, "${base_dir_REAL}/webapp")
-  $graphiteweb_storage_dir_REAL     = $gr_graphiteweb_storage_dir
+  $graphiteweb_storage_dir_REAL     = pick($gr_graphiteweb_storage_dir, $gr_graphiteweb_storage_dir)
```
This makes no sense and the rspec checks are failing.

JoshiiSinfield replied:
sorry - mistake there. should have been $storage_dir_REAL... as config.pp was using $storage_dir_REAL as the directory for graphite.db. My thinking was that if you don't specify a custom location to the $graphiteweb_storage_dir then use $storage_dir_REAL.

Does that sound ok? if so I'll amend the commit and push back up.

JoshiiSinfield commented:
Hi @dwerder 

I've removed my changes from init.pp as I got confused with what I was actually trying to achieve in there... and it's raised a question around the rspecs (of which I've never been able to get running on my work laptop - windows. I'll spin up a vagrant machine to run the tests instead.)

You're testing specifically for opt/graphite/storage/graphite.db to exist, which now that I've changed config.pp to set the permissions on the correct file no longer exists. BUT the module was always set to use $gr_graphiteweb_storage_dir as the value for $graphiteweb_storage_dir_REAL in init.pp, which had a default to /var/lib/graphite-web.

So - my question is: do we need to do a pick for graphiteweb_storage_dir_REAL between gr_graphiteweb_storage_dir & $storage_dir_REAL or do we need to change the tests to check for the file existing & permissions in the location passed in to the parameter?

Cheers,
Josh

walkamongus commented:
+1

I just ran into one of the problems addressed by these fixes. The graphite-web DB file was created with incorrect permissions so Apache couldn't access it when starting.

The module also tried to initialize the DB before the storage directory exists due to the file resource that sets up the storage directory subscribing to the syncdb exec at  https://github.com/echocat/puppet-graphite/blob/master/manifests/config.pp#L121
